server/atlassian_connect.go

```diff
@@ -50,6 +51,26 @@ func httpACInstalled(p *Plugin, w http.ResponseWriter, r *http.Request) (int, er
 			errors.WithMessage(err, "failed to unmarshal request")
 	}
 
+	// Only allow this operation once, a JIRA instance must already exist
+	// for asc.BaseURL but its EventType would be empty.
+	ji, err := p.LoadJIRAInstance(asc.BaseURL)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to load instance "+asc.BaseURL)
+	}
+	if ji == nil {
+		return http.StatusNotFound,
+			errors.New(fmt.Sprintf("Jira instance %s must first be added to Mattermost", asc.BaseURL))
+	}
+	jci, ok := ji.(*jiraCloudInstance)
+	if !ok {
+		return http.StatusBadRequest, errors.New("Must be a JIRA Cloud instance, is " + ji.GetType())
+	}
+	if jci.EventType == "installed" {
```

Comment:
This just seems like a strange thing to check here. Maybe something dedicated to this purpose instead of storing a blank instance?

Reply:
Now an `Installed` flag in jci